BrownianMotion/StochasticIntegral/Predictable.lean

```diff
@@ -0,0 +1,217 @@
+/-
+Copyright (c) 2025 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+import Mathlib.Probability.Process.Filtration
+import Mathlib.Probability.Process.Adapted
+
+/-!
+# Progressively Measurable σ-algebra
+
+This file defines the progressively measurable σ-algebra associated to a filtration, as well as the
+notion of predictable processes. We prove that predictable processes are progressively measurable
+and adapted. We also give an equivalent characterization of predictability for discrete processes.
+
+## Main definitions
+* `Filtration.Predictable` : The predictable σ-algebra associated to a filtration.
+* `Filtration.IsPredictable` : A process is predictable if it is measurable with respect to the
+  predictable σ-algebra.
+
+## Main results
+* `Filtration.IsPredictable.progMeasurable` : A predictable process is progressively measurable.
+* `Filtration.IsPredictable.measurable_succ` : `u` is a discrete predictable process iff
+  `u (n + 1)` is `𝓕 n`-measurable and `u 0` is `𝓕 0`-measurable.
+
+-/
+
+open Filter Order TopologicalSpace
+
+open scoped MeasureTheory NNReal ENNReal Topology
+
+namespace MeasureTheory
+
+variable {Ω ι : Type*} {m : MeasurableSpace Ω} {E : Type*} [TopologicalSpace E]
+
+section
+
+variable [Preorder ι]
+
+/-- A filtration `𝓕` is right continuous if `𝓕 t = ⨅ j > i, 𝓕 j = 𝓕 i` for all `t`. -/
+class RightContinuous (𝓕 : Filtration ι m) where
```

Comment:
Let's use an `Is` prefix for `Prop` defs, unless it's too weird grammatically (`IsUsualConditions` would be too weird imo).
We did not follow that rule for `Adapted`, `Martingale` and all the defs we added back then, but I think that's something we should change eventually to better match latest Mathlib practice.
```suggestion
class IsRightContinuous (𝓕 : Filtration ι m) where
```